### DISCUSSION TITLE
Docusaurus basic auth with edge middleware

### DISCUSSION THREAD

**Discussion by jiri-dxheroes** [community/community#68358] [Programming Help] [answered]
### Body

Hey, I'm trying to get a basic auth using authorization header working in Docusaurus@2.4.1, but can't seem to figure it out. I tried implementing [this](https://github.com/vercel/examples/blob/main/edge-middleware/basic-auth-password/middleware.ts) vercel example, which worked for me in Nextjs, but not in Docusaurus. Vercel says it works with every framework, so I am probably doing something wrong but don't know what. This is the code I got to so far:
```
import { next } from '@vercel/edge';

export const config = {
  matcher: ['/'],
};

export function middleware(req: Request) {
  const basicAuth = req.headers.get('authorization');

  if (basicAuth) {
    const authValue = basicAuth.split(' ')[1];
    const [user, pwd] = atob(authValue).split(':');

    if (user === '4dmin' && pwd === 'testpwd123') {
      return next();
    }
  }

  return new Response('Authentication required!', {
    status: 401,
    headers: {
      'WWW-Authenticate': "Basic realm='Secure Area'",
    },
  });
}
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by jsimecek** [comment, marked as answer]
You are missing a default export, which Docusaurus for some reason requires. I had the same issue recently and spent like 4 hours just on this 🙃
